telemetry-aware-scheduling/deploy/cluster-api/docker/capi-docker.md

```diff
@@ -0,0 +1,160 @@
+# Cluster API deployment - Docker provider (for local testing/development only)
+
+## Requirements
+
+- A management cluster provisioned in your infrastructure of choice and the relative tooling.
+  See [Cluster API Quickstart](https://cluster-api.sigs.k8s.io/user/quick-start.html).
+- Run Kubernetes v1.22 or greater (tested on Kubernetes v1.25).
+- Docker
+
+## Provision clusters with TAS installed using Cluster API
+
+We will provision a KinD cluster with the TAS installed using Cluster API. This guide is meant for local testing/development only.
+
+For the deployment using a generic provider, please refer to [Cluster API deployment - Generic provider](capi.md).
+
+1. Run the following to set up a KinD cluster for CAPD:
+
+```bash
+cat > kind-cluster-with-extramounts.yaml <<EOF
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+- role: control-plane
+  extraMounts:
+    - hostPath: /var/run/docker.sock
+      containerPath: /var/run/docker.sock
+EOF
+```
+
+2. Enable the `CLUSTER_TOPOLOGY` feature gate:
+
+```bash
+export CLUSTER_TOPOLOGY=true
+```
+
+3. Initialize the management cluster:
+
+```bash
+clusterctl init --infrastructure docker
+```
+
+Run the following to generate the default cluster manifests:
+
+```bash
+clusterctl generate cluster capi-quickstart --flavor development \
+  --kubernetes-version v1.25.0 \
+  --control-plane-machine-count=3 \
+  --worker-machine-count=3 \
+  > capi-quickstart.yaml
+```
+
+Be aware that you will need to install a CNI such as Calico before the cluster will be usable. You may automate this
```

Comment:
For the sake of simplicity I decided to add the CRS for Calico, this way it will auto-install. 
I removed this line but added a notice in the generic provider guide since Calico might not be needed/compatible with some providers. There's no need to such notice for the Docker provider since only Calico is guaranteed to work by the maintainers.